thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/cache/RedisCacheDAO.java

```diff
@@ -0,0 +1,273 @@
+import org.redisson.api.RTimeSeries;
+import org.redisson.api.RedissonClient;
+import org.redisson.client.RedisConnectionException;
+import org.redisson.config.Config;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * DAO class used to fetch data from Redis.
+ */
+
+public class RedisCacheDAO implements CacheDAO {
+
+	private static final String HOSTS = "hosts";
+	private static final String AUTH_USERNAME = "authUsername";
+	private static final String AUTH_PASSWORD = "authPassword";
+	public static final String IDLE_CONNECTION_TIMEOUT = "idleConnectionTimeout";
+	public static final String CONNECT_TIMEOUT = "connectTimeout";
+	public static final String TIMEOUT = "timeout";
+	public static final String RETRY_ATTEMPTS = "retryAttempts";
+	public static final String RETRY_INTERVAL = "retryInterval";
+	public static final String PING_CONNECTION_INTERVAL = "pingConnectionInterval";
+	public static final String KEEP_ALIVE = "keepAlive";
+	public static final String TCP_NO_DELAY = "tcpNoDelay";
+
+	public static final String DEFAULT_HOST = "redis://127.0.0.1:6379";
+	public static final String DEFAULT_AUTH_USERNAME = "";
+	public static final String DEFAULT_AUTH_PASSWORD = "";
+	public static final int DEFAULT_IDLE_CONNECTION_TIMEOUT = 10000;
+	public static final int DEFAULT_CONNECT_TIMEOUT = 10000;
+	public static final int DEFAULT_TIMEOUT = 3000;
+	public static final int DEFAULT_RETRY_ATTEMPTS = 3;
+	public static final int DEFAULT_RETRY_INTERVAL = 1500;
+	public static final int DEFAULT_PING_CONNECTION_INTERVAL = 30000;
+	public static final boolean DEFAULT_KEEP_ALIVE = false;
+	public static final boolean DEFAULT_TCP_NO_DELAY = false;
+
+	// private static final String SERIALIZATION = "serialization";
+	// public static final String DEFAULT_SERIALIZATION = "json";
+	// public static final String DEFAULT_ALLOWED_TYPE = "aln,var";
+
+	private static final Logger LOG = LoggerFactory.getLogger(RedisCacheDAO.class);
+
+	private Config redissonConfig;
+	private RedissonClient redissonClient;
+	private boolean redisState;
+
+	// Eg: KEY = metricId:2:dimensionKey:3158902058
+	private static final String KEY_PREFIX = "metricId:";
+
+	public RedisCacheDAO() {
+		this.createDataStoreConnection();
+	}
+
+	/**
+	 * Initialize connection to Redis
+	 */
+	private void createDataStoreConnection() {
+		LOG.info("Starting Redis Connection......");
+		CacheDataSource dataSource = CacheConfig.getInstance().getCentralizedCacheSettings().getDataSourceConfig();
+		Map<String, Object> dataSourceconfig = dataSource.getConfig();
+		List<String> hosts = ConfigUtils.getList(dataSourceconfig.get(HOSTS));
+		redissonConfig = new Config();
+
+		// Taking only one host as of now.
+		String host = "";
+		if (hosts.size() == 0) {
+			host = DEFAULT_HOST;
+		} else {
+			host = (String) hosts.toArray()[0];
+		}
+		redissonConfig.useSingleServer().setAddress(host);
+
+		// If no username specified, default username ("") will be considered
```

Comment:
The default username/password is non empty. Is this guaranteed to work without this check here?

Reply:
uncommented the req code